ledger/acctdeltas.go

```diff
@@ -1011,21 +1027,12 @@ func onlineAccountsNewRoundImpl(
 							Ref:         ref,
 							UpdRound:    basics.Round(updRound),
 						}
+
 						updatedAccounts = append(updatedAccounts, updated)
 						prevAcct = updated
-					} else if !newAcct.IsVotingEmpty() {
-						err = fmt.Errorf("non-empty voting data for non-online account %s: %v", data.address.String(), newAcct)
-						return nil, err
```

Comment:
`Online` can have voting data, and now `Offline` can too, as you noted.

I don't see a way for `NotParticipating` to have voting data, though this PR would indeed stop noticing if we had a bug that caused that to happen.

The newly introduced transactions are:
1. Suspend - which moves from Offline to Online and retains voting data.
2. Heartbeat - which can move from Offline to Online, iff the Offline account has voting data to heartbeat with.

I suppose a thing for me to test is that going from Offline to NP clears voting data. In the past there would be no need to do so, but I suspect that transition will blank the voting data because it will only succeed when the txn has blank voting data, and it copies that blankness into the account without regard to whether it was coming "from `Online` or `Offline`.  Will confirm.

Reply:
It doesn't actually seem like such a horrific bug if it were to happen.  The old voting data would remain stored. I guess that's not intended, but does it really matter so much?  You can never leave NP.

Reply:
+1 lets do that

Reply:
I've confirmed that going NonPart requires empty keys, and is tested in:
`func TestWellFormedKeyRegistrationTx(t *testing.T) {`

And then I've confirmed that those empty keys are used to wipeout any voting data that might be there because the account was suspended.

I assume that the idea of introducing invariant checks in `endOfBlock` is intended for another PR, but I will let Chris or Pavel resolve this if they agree.